services/network_config_manager/src/main/java/com/futurewei/alcor/netwconfigmanager/cache/HostResourceStateCache.java

```diff
@@ -0,0 +1,42 @@
+package com.futurewei.alcor.netwconfigmanager.cache;
+
+import com.futurewei.alcor.common.db.CacheFactory;
+import com.futurewei.alcor.common.db.ICache;
+import com.futurewei.alcor.common.stats.DurationStatistics;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@ComponentScan(value = "com.futurewei.alcor.common.db")
+public class HostResourceStateCache {
+    // Map <ResourceId, ResourceState>
+    private ICache<String, Object> hostResourceStates;
```

Comment:
@xieus Is HostResourceStatesCache better?

Reply:
Is it correct to state that HostResourceStateCache entires are connected to HostResourceIdCache through the key (ResourceId) of HostResourceStateCache? Without this clarification, it is possible to assume that since HostResourceIdCache  has hostid as the key, HostResourceStateCache may also have the same key.

Reply:
> @xieus Is HostResourceStatesCache better?

Hmm, we can justify either of them :-) For class name, I prefer singular. What do you think of?

Reply:
I prefer singular in most cases and this is one of them.

Reply:
> Is it correct to state that HostResourceStateCache entires are connected to HostResourceIdCache through the key (ResourceId) of HostResourceStateCache? 

You are right. 

> Without this clarification, it is possible to assume that since HostResourceIdCache has hostid as the key, HostResourceStateCache may also have the same key.

Hmm, good point. I think we should rename HostResourceStateCache to ResourceStateCache to avoid confusion.

Reply:
ResourceStateCache fits better.

Reply:
Changed.